framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -1481,7 +1481,12 @@ public Void visitUnary(UnaryTree node, Void p) {
      */
     @Override
     public Void visitCompoundAssignment(CompoundAssignmentTree node, Void p) {
-        commonAssignmentCheck(node.getVariable(), node.getExpression(),
+        // Given a compound assignment:
+        // s += expr;
+        // Should be whether s + expr can be assigned to s,
+        // but the "s + expr" tree does not exist.  So instead, check that
+        // s += expr can be assigned to s.
+        commonAssignmentCheck(node.getVariable(), node,
```

Comment:
Recommend making it extra clear in the comment that `s += expr` is `node` in this case. This is very easy to understand when seeing the diff, but once the diff is gone, it may not be immediately obvious.